src/wh_client_cryptocb.c

```diff
@@ -594,6 +585,57 @@ int wolfHSM_CryptoCb(int devId, wc_CryptoInfo* info, void* inCtx)
         }
         break;
 #endif /* !WC_NO_RNG */
+#ifdef WOLFSSL_CMAC
+    case WC_ALGO_TYPE_CMAC:
+        /* in, key and out are after the fixed size fields */
+        in = (uint8_t*)(&packet->cmacReq + 1);
+        key = in + info->cmac.inSz;
+        out = (uint8_t*)(&packet->cmacRes + 1);
+        packet->cmacReq.type = info->cmac.type;
+        packet->cmacReq.keyId = (intptr_t)info->cmac.cmac->devCtx;
+        /* multiple modes are possible so we need to set zero size if buffers
+         * are NULL */
```

Comment:
Currently, key!=NULL && keySz!=0 means to invoke Init.  in!=NULL && inSz!=0 means to invoke Update.  out!=NULL && outSz!=0 means to invoke Finalize (which should clear the server context).

Seems like the incremental path using a keyId has an issue as I would expect the client to be able to:
```
Cmac cmacctx[1] = {0};
wc_Init_Cmac_ex(cmacctx, NULL, 0, WC_CMAC_AES, NULL, NULL, WOLFHSM_DEV_ID);
wh_SetCmacKey(cmacctx, MY_KEY_ID);  //This helper function is missing
wc_CmacUpdate(cmacctx, data, datasize); //Server MUST keep the server context around OR must pass back the partial state to the client
wc_CmacFinalNoFree(cmacctx, mac, mac size);  //Server ALWAYS clears the server context on finalize.  The free/nofree is for the local context
wc_CmacFree(cmacctx);
```